### DISCUSSION TITLE
Proposal: "Community" releases with a lower release cadence

### DISCUSSION THREAD

**Discussion by JonasHelming** [eclipse-theia/theia#11047] [Improvements] [answered]
This is a proposal extracted from the [discussion on how to make it easier to consume/adopt Theia](https://github.com/eclipse-theia/theia/discussions/10791).

The proposal is to mark specific releases, e.g. quartely as "community" releases. Adopters who do not want to adopt releases on a monthly basis can chose to adopt the community releases to create synergies (details see below). The monthly release cycle remains unchanged.

Theia follows a monthly release cycle and we do not want to change this to continuously drive innovation. Some adopting projects adopt a new release on a monthly basis and will continue to do so.
However, some other adopting projects do not adopt every monthly release. This is especially the case for tools based on Theia that are deployed in production or to end users and therefore have a slower release cycle compared to Theia. These projects currently  select arbitrary Theia versions to adopt and skip some other versions. With the introduction of community releases, adopters could chose these versions, it is more likely that more projects adopt the same version.

Having multiple adopter projects adopting the same Theia release would potentially enable synergies, though. As an example, if a security fix is necessary, an adopting project could contribute a patch to an older community release. Other adopters on the same release could benefit from this.

Another aspect are frameworks that integrate with Theia, e.g. Eclipse GLSP, EMF.cloud or CDT.cloud. Ensuring compatibility with every new monthly Theia release is quite a lot of work. Integrating frameworks could chose to validate their integration against community releases  and communicate this to their users. Adopters of the ecosystem would more likely find a set of compatible technologies using a community release.

From a projects point of view, the minimum support for this proposal would be to declare specific releases as community releases. There are currently no plans to explicitly maintain these releases, any patches an re-releases would need to be contributed or sponsored.
The project could additionally decide to avoid high risk changes in community releases to improve their stability.

Again, the monthly release cycle would remain unchanged.

Two parameters are open for this proposal:

The frequency (e.g. quartely, every 4 months or twice a year( and the name for community release (e.g. "community", "baseline", "foobar").
Please comment:
-  whether you believe that this proposal would be beneficial for adopters
- What frequency you prefer
- Name suggestions for this type of release

Please also forward this to projects adopting Theia that you are aware of

**Comment by Niklas-Dahlquist** [comment]
* Good initiative, I believe that having community releases will be focused for a good stability and will get more confidence in the development on the Theia Framework.

* Since the evolution of Theia is high cause of a lot of committed users, for now I suggest quarterly release cycle, to allow to follow the evolution but not have too many breaking changes to comply on.

* ```community``` releases sounds good since it's not exactly a LTS suggestion

**Comment by colin-grant-work** [comment]
I think this would work best if combined with a very limited probationary period, so that in the month following the `latest` release that will form the basis of the `community` release, we apply any critical patches to a separate branch, and at the end of that month, that branch becomes the basis of the `community` release. That way, we don't have to modify the process for the `latest` release unduly - delaying or selecting PR's that should or shouldn't make it into the `community` release - but we would be sure that the `community` release is reasonably stable and bug free without a full commitment to LTS. Perhaps @vince-fugnitto  can weigh in on the difficulties associated with integrating that approach into our release process.

**Reply by JonasHelming** [reply]
@vince-fugnitto ?

**Reply by vince-fugnitto** [reply]
I think it's fine to have community releases (we can discuss the frequency). 

In order to reduce the overhead, I would suggest that others help out for such releases so I'm not the only one that needs to do this overhead. I can be the one that releases it however, but applying patches and managing the release branch can be any contributor.

**Reply by JonasHelming** [reply]
fully agreed!

**Comment by JonasHelming** [comment, marked as answer]
The idea of a community release has been introduced: https://theia-ide.org/releases/
